sql/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java

```diff
@@ -87,7 +91,8 @@ private static SysSnapshot createSysSnapshot(Repository repository, SnapshotId s
             snapshotInfo.startTime(),
             snapshotInfo.endTime(),
             version == null ? null : version.toString(),
-            snapshotInfo.state().name()
+            snapshotInfo.state().name(),
+            Lists2.map(snapshotInfo.shardFailures(), SnapshotShardFailure::toString)
```

Comment:
This would result in an array with items like:

```
SnapshotShardFailure{shardId=[.partitioned.foobar.04732d9n60q30dhk60o30c1g][1], reason='IndexShardSnapshotFailedException[Failed to get store file metadata]; nested: CorruptIndexException[verification failed : calculated=dvytoi stored=0 (resource=VerifyingIndexInput(NIOFSIndexInput(path=\"/mnt/data10/data/nodes/0/indices/iEVz27NqTHC7ryzS9bxosg/1/index/_1jxq.fdt\")))]; ', nodeId='rqczRQwdTwqSK34DalJS4w', status=INTERNAL_SERVER_ERROR}
```

Should we instead go for an array object ?  where the nodeId/index/shardId is available separate as well?

Reply:
Not sure if this is worth it/needed as I think this is mostly interesting by manual/human debugging.